keyboards/handwired/stef9998/split_5x7/keymaps/stef9998/keymap.c

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2022 stef9998
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// You can also skip layer-names entirely and just use numbers.
+#define _QWERTY 0
+#define _HOMEMOD 2
+#define _FN 3
+#define _SYM 4
+#define _SPARE 6
+#define _CFG 7//config
+
+enum custom_keycodes {
+    QWERTY = SAFE_RANGE,
+    HOMEMOD,
+    FN,
+    SYM,
+    SPARE,
+    CONFIG,
+};
+
+// Shortcut to make keymap more readable
+#define KC_FN MO(_FN)
+
+bool get_ignore_mod_tap_interrupt(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case KC_LSPO:
+            // Do not force the mod-tap key press to be handled as a modifier
+            // if any other key was pressed while the mod-tap key is held down.
+            return false;
+//        case MT(MOD_LSFT,KC_9):
+//            return false;
+        case KC_RSPC:
+            return false;
+        default:
+            // Force the mod-tap key press to be handled as a modifier if any
+            // other key was pressed while the mod-tap key is held down.
+            return true;
+    }
+}
+
+//layer_state_t layer_state_set_user(layer_state_t state) {
+//    return update_tri_layer_state(state, _ESS, _SYM, _SYM_ESS);
+//}
+
+const keypos_t PROGMEM hand_swap_config[MATRIX_ROWS][MATRIX_COLS] = {
+{ {0, 5},  {1, 5},  {2, 5},  {3, 5}, {4, 5}, {5, 5}, {6, 5}},
+{ {0, 6},  {1, 6},  {2, 6},  {3, 6}, {4, 6}, {5, 6}, {6, 6}},
+{ {0, 2},  {1, 7},  {2, 7},  {3, 7}, {4, 7}, {5, 7}, {6, 7}}, // caps won't be switched, because of extend
+{ {0, 8},  {1, 8},  {2, 8},  {3, 8}, {4, 8}, {5, 8}, {6, 8}},
+{ {0, 9},  {1, 9},  {2, 9},  {3, 9}, {4, 9}, {5, 9}, {6, 9}},
+
+{ {0, 0},  {1, 0},  {2, 0},  {3, 0}, {4, 0}, {5, 0}, {6, 0}},
+{ {0, 1},  {1, 1},  {2, 1},  {3, 1}, {4, 1}, {5, 1}, {6, 1}},
+{ {0, 2},  {1, 2},  {2, 2},  {3, 2}, {4, 2}, {5, 2}, {6, 2}},
+{ {0, 3},  {1, 3},  {2, 3},  {3, 3}, {4, 3}, {5, 3}, {6, 3}},
+{ {0, 4},  {1, 4},  {2, 4},  {3, 4}, {4, 4}, {5, 4}, {6, 4}},
+};
```

Comment:
This should be placed in the keyboard's c file, not here. 

If it needs to be customized, then both, with it set with `__attribute__((weak))` in the keyboard's version. 
```suggestion
```

Reply:
So I understand it right, that even if it is not used in the default keymap, it should still be included in the keyboards main c file?

Reply:
I just looked up, how it's implemented on a let's split after a quick websearch. It is included in the keyboard's c file by using `#ifdef`'s obviously. That explaines it for me, and my question is resolved^^